datastore/dsmodels/builder.go

```diff
@@ -155,19 +155,21 @@ func (b *builder[C, T, M]) First(ctx context.Context) (M, error) {
 }
 
 func (b *builder[C, T, M]) Get(ctx context.Context) ([]M, error) {
-	items := []M{}
+	itemPtrs := []*M{}
```

Comment:
I would initialize the slice with the correct size: 

`make([]*M,0,len(b.ids)`

Or:

```go
itemPtrs := make([]*M,len(b.ids)
for i, id := range b.ids {
  itemPtrs[i] = b.value.lazy(...)
}
```